src/keycloak/chart/templates/keycloak-logging-recording-rules.yaml

```
@@ -0,0 +1,29 @@
+# Copyright 2025 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+{{- if .Values.detailedObservability }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "keycloak.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    loki_rule: "1"
+data:
+  recording-rules.yaml: |
+    groups:
+      - name: keycloak-change-metrics
+        interval: 30s
+        rules:
+          - record: uds_keycloak:realm_modifications
```

```suggestion
          - record: uds_keycloak:realm_modifications_total
```
_total indicates a counter metric. It is best practice to suffix this with a _total https://prometheus.io/docs/practices/naming/#metric-names